packages/modules/order/src/services/actions/cancel-return.ts

```diff
@@ -0,0 +1,98 @@
+import {
+  Context,
+  CreateOrderChangeActionDTO,
+  OrderTypes,
+} from "@medusajs/types"
+import { promiseAll } from "@medusajs/utils"
+import { ChangeActionType } from "../../utils"
+
+async function createOrderChange(
+  service,
+  data,
+  returnRef,
+  actions,
+  sharedContext
+) {
+  return await service.createOrderChange_(
+    {
+      order_id: returnRef.order_id,
+      return_id: returnRef.id,
+      reference: "return",
+      reference_id: returnRef.id,
+      description: data.description,
+      internal_note: data.internal_note,
+      created_by: data.created_by,
+      metadata: data.metadata,
+      actions,
+    },
+    sharedContext
+  )
+}
+
+export async function cancelReturn(
+  this: any,
+  data: OrderTypes.CancelOrderReturnDTO,
+  sharedContext?: Context
+) {
+  const returnOrder = await this.retrieveReturn(
+    data.return_id,
+    {
+      select: [
+        "id",
+        "order_id",
+        "items.item_id",
+        "items.quantity",
+        "items.received_quantity",
+        "items.item.title",
+        "reason.id",
+      ],
+      relations: ["items"],
+    },
+    sharedContext
+  )
+
+  const actions: CreateOrderChangeActionDTO[] = []
+
+  returnOrder.items.forEach((item) => {
```

Comment:
question: Do we need to think about canceling shipping?

Reply:
yes, absolutely.
Good point.